components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -445,23 +443,6 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
                 `operation not permitted: missing ${op} permission on ${resource.kind}`,
             );
         }
-        if (resource.kind === "workspace" && op === "get") {
-            // access to workspaces is granted. Let's verify that this is also thecase with FGA
-            const result = await this.auth.hasPermissionOnWorkspace(
-                this.userID!,
-                "read_info",
-                resource.subject.id,
-                true,
-            );
-            if (!result) {
-                const isShared = await this.auth.find(rel.workspace(resource.subject.id).shared.anyUser);
-                log.error("user has access to workspace, but not through FGA", {
```

Comment:
We have a better check in authorizer that verifies the `shared` relationship exists after it was written.